Description: fix makealiases .xalias file creation
 With legacy sendmail aliases to programs or mailboxes makealiases
 creates dummy dot-courier files in a special directory. These were
 misnamed, however.
Author: Markus Wanner <markus@bluegap.ch>
Bug-Debian: http://bugs.debian.org/703570
Forwarded: no
Last-Update: 2013-03-21
--- a/courier/aliasexp.C
+++ b/courier/aliasexp.C
@@ -128,13 +128,17 @@
 
 	std::string s=aliasnameEscaped.str();
 
+	std::string defaultDomainEscaped = config_defaultdomain();
+	std::replace(defaultDomainEscaped.begin(),
+		defaultDomainEscaped.end(), '.', ':');
+
 	if (first_time)
 	{
 		first_time=0;
 		mkdir(tmpdir, 0755);
 	}
 
-	std::string t=std::string(tmpdir) + "/" + s;
+	std::string t=std::string(tmpdir) + "/" + s + '@' + defaultDomainEscaped;
 
 	std::cout << "<\"" << xpfix << s
 		  << "\"@" << config_defaultdomain() << ">" << std::endl;
